crates/re_ui/src/list_item2/scope.rs

```diff
@@ -34,9 +34,12 @@ pub struct State {
     ///
     /// The semantics are exactly the same as for `left_column_width`.
     max_desired_left_column_width: f32,
-    /**/
-    // TODO(#6179): record the use of right action button in all PropertyContent such as to not
-    // unnecessarily reserve right gutter space if none have it.
+
+    /// If true, right-aligned space should be reserved for the action button, even if not used.
+    pub(crate) reserve_action_button_space: bool,
+
+    /// Track whether the action button space should be reserved.
+    should_reserve_action_button_space: bool,
```

Comment:
I find it confusing that we have two bool for this.

The whole `State` is a bit confusing.

There are, I think, three categories of things in `State`:

* Fixed settings: Things set once `list_item_scope` and never changed
* Frame accumulators: Things reset at the start of `list_item_scope` and _written_ to by children, then stored to be used the next frame (e.g. desired column widths)
* Memory: things we remember from previous frames

We should make these categories much clearer, e.g. by grouping them inside of `State` and commenting it

Reply:
Well, to me it made more sense to group things by functionality rather than accumulator vs. something else.

Also, your categories are not quite right. There are (currently) two things:
- Accumulators that saved to memory and read the next frame, and then reset again for accumulation.
- Fixed values set by the scope once (either from external sources or from the last-frame accumulated values).

Anyways, I'll try to clarify this. Maybe it can be split in two structs to make things more explicit.

Reply:
Okay, I ended up doing a pretty significant overhaul of the whole thing.

`State` is gone. Now we have a  (private) `LayoutStatistics` struct that accumulates layout "wants" during the frame. It's stored in egui's memory against the scope id. Then we have the (public) `LayoutInfo` which contains the layout information + other things that a prepared once by the `list_item_scope` and provided read-only to the child list items. For convenience, `LayoutInfo` _also_ provides the public API used to gather the information that ends up in `LayoutStatistics`.

TL;DR: the API for user code (aka ListItemContent impls) is cleaner, and the implementation is more correct and, I think, much less confusing.